zerver/lib/integrations.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, name: str, client_name: str, categories: List[str],
                  logo: Optional[str]=None, secondary_line_text: Optional[str]=None,
                  display_name: Optional[str]=None, doc: Optional[str]=None,
                  stream_name: Optional[str]=None, legacy: bool=False,
-                 config_options: Optional[List[Tuple[str, str, Validator]]]=None) -> None:
+                 config_options: List[Tuple[str, str, Validator]]=[]) -> None:
```

Comment:
Are we opening ourselves up to this pitfall?

~~~ py
  1 def add(info, lst=None):
  2     if lst is None:
  3         lst = []
  4     lst.append(info)
  5     print(lst)
  6
  7 def add2(info, lst=[]):
  8     lst.append(info)
  9     print(lst)
 10
 11
 12 add('a')
 13 add('b')
 14
 15
 16 add2('c')
 17 add2('d')
~~~

output:

~~~
['a']
['b']
['c']
['c', 'd']

Reply:
We can guard against that by replacing `List` with the read-only `Sequence` type. I’ll separate this out.